drivers/hmis/spec/models/hmis/form/form_processor_spec.rb

```diff
@@ -1801,6 +1801,45 @@ def add_cde_item_to_definition
           expect(record.custom_data_elements.map(&:date_updated)).not_to include(old1.date_updated, old2.date_updated)
         end
 
+        it 'does not delete and replace when some values are the same, but some are different (repeats: true)' do
+          record = i1
+          old_cded = create(:hmis_custom_data_element, owner: record, value_string: 'old value', data_element_definition: cded)
+          expect(record.custom_data_elements.size).to eq(1)
+
+          hud_values = complete_hud_values.merge(
+            cded.key => [old_cded.value_string, 'new value'],
+          )
+
+          expect do
+            process_record(record: record, hud_values: hud_values, user: hmis_user, definition: definition)
+            record.reload
+            old_cded.reload
+          end.to change(old_cded, :date_updated).and not_change(old_cded, :date_deleted)
+
+          expect(record.custom_data_elements.size).to eq(2)
+          expect(record.custom_data_elements.map(&:id)).to include(old_cded.id)
+        end
+
+        it 'behaves correctly when given identical values' do
+          record = i1
+
+          hud_values = complete_hud_values.merge(
+            cded.key => ['foo', 'foo'],
+          )
+          process_record(record: record, hud_values: hud_values, user: hmis_user, definition: definition)
+
+          expect(record.custom_data_elements.size).to eq(2)
+          expect(record.custom_data_elements.map(&:value_string)).to contain_exactly('foo', 'foo')
+
+          hud_values = complete_hud_values.merge(
+            cded.key => ['foo', 'foo', 'foo', 'bar'],
+          )
+          process_record(record: record, hud_values: hud_values, user: hmis_user, definition: definition)
+
+          # expect(record.custom_data_elements.size).to eq(4)
```

Comment:
The commented-out line fails, since `custom_data_elements.size` is equal to 3. The last `foo` value doesn't get properly added because it's considered an existing value.

I _think_ the most graceful way to deal with this is to just prevent submission of duplicate values. That way we don't have to keep track of value counters or anything like that. Maybe in `normalize_custom_field_value`?

Reply:
I implemented this, let me know what you think